typescript/radix-sdk/src/modules/tx.ts

```diff
@@ -0,0 +1,519 @@
+      from_address: this.account.address,
+      token,
+      destination_domain,
+      recipient,
+      amount,
+      custom_hook_id,
+      gas_limit,
+      custom_hook_metadata,
+      max_fee,
+    });
+
+    await this.signAndBroadcast(transactionManifest);
+  }
+
+  public async signAndBroadcast(
+    manifest: TransactionManifest,
+  ): Promise<TransactionHash> {
+    const constructionMetadata =
+      await this.gateway.transaction.innerClient.transactionConstruction();
+
+    const transactionHeader: TransactionHeader = {
+      networkId: this.networkId,
+      startEpochInclusive: constructionMetadata.ledger_state.epoch,
+      endEpochExclusive: constructionMetadata.ledger_state.epoch + 2,
+      nonce: generateRandomNonce(),
+      notaryPublicKey: this.account.publicKey,
+      notaryIsSignatory: true,
+      tipPercentage: 0,
+    };
+
+    const transaction: NotarizedTransaction =
+      await TransactionBuilder.new().then((builder) =>
+        builder
+          .header(transactionHeader)
+          .manifest(manifest)
+          .sign(this.signIntent)
+          .notarize(this.notarizeIntent),
+      );
+
+    const compiledNotarizedTransaction =
+      await RadixEngineToolkit.NotarizedTransaction.compile(transaction);
+
+    const intentHashTransactionId =
+      await RadixEngineToolkit.NotarizedTransaction.intentHash(transaction);
+
+    await this.gateway.transaction.innerClient.transactionSubmit({
+      transactionSubmitRequest: {
+        notarized_transaction_hex: Buffer.from(
+          compiledNotarizedTransaction,
+        ).toString('hex'),
+      },
+    });
+    await this.query.pollForCommit(intentHashTransactionId.id);
+
+    return intentHashTransactionId;
+  }
+
+  public async getTestnetXrd() {
+    const constructionMetadata =
+      await this.gateway.transaction.innerClient.transactionConstruction();
+
+    const freeXrdForAccountTransaction =
+      await SimpleTransactionBuilder.freeXrdFromFaucet({
+        networkId: this.networkId,
+        toAccount: this.account.address,
+        validFromEpoch: constructionMetadata.ledger_state.epoch,
+      });
+
+    const intentHashTransactionId =
+      freeXrdForAccountTransaction.transactionId.id;
+
+    await this.gateway.transaction.innerClient.transactionSubmit({
+      transactionSubmitRequest: {
+        notarized_transaction_hex: freeXrdForAccountTransaction.toHex(),
+      },
+    });
+    await this.query.pollForCommit(intentHashTransactionId);
+
+    return intentHashTransactionId;
+  }
+
+  private async getNewComponent(transaction: TransactionHash): Promise<string> {
+    const transactionReceipt =
+      await this.gateway.transaction.getCommittedDetails(transaction.id);
+
+    const receipt = transactionReceipt.transaction.receipt;
+    assert(receipt, `found no receipt on transaction: ${transaction.id}`);
+
+    const newGlobalGenericComponent = (
+      receipt.state_updates as any
```

Comment:
Is there any way we can avoid the any here? This type assertion could hide errors in case we upgrade the radix library to a newer version and there is a breaking change

Reply:
Unfortunately some properties just have the type "object" which says nothing, will add custom types instead of any